### DISCUSSION TITLE
New user trouble

### DISCUSSION THREAD

**Discussion by jason-cso** [pymodbus-dev/pymodbus#1537] [Help]
It seems like you've tried to make pymodbus easy to get started with, but I've had nothing but trouble.

I used pip to install pymodbus 3.2.2 and tried:  `$ pymodbus.console serial --port /dev/ttyUSB0 --baudrate 9600 --timeout 2`, but a bunch of the client.* options seem to be missing.  For example, there's no client.connect

Next, I tried `client_sync.py` from github as the readme suggests since I didn't seem to have the examples locally, but it fails with `No module named 'examples.helper'`.  I tried pip installing pymodbus[documentation] but that just installed a bunch of sphinx dependencies I didn't want.

Next I tried this:
```
#!/usr/bin/env python3.8

import pymodbus
from pymodbus.client import ModbusSerialClient

client = ModbusSerialClient(
    port="/dev/ttyUSB0",
    baudrate=9600,
)

client.connect()
result = client.read_coils(1, 1)
```

But that just gives me:

```
Traceback (most recent call last):
  File "./foo.py", line 12, in <module>
    result = client.read_coils(1, 1)
  File "/home/j/.local/lib/python3.8/site-packages/pymodbus/client/mixin.py", line 72, in read_coils
    return self.execute(
  File "/home/j/.local/lib/python3.8/site-packages/pymodbus/client/base.py", line 197, in execute
    raise ConnectionException(f"Failed to connect[{str(self)}]")
pymodbus.exceptions.ConnectionException: Modbus Error: [Connection] Failed to connect[ModbusSerialClient(<pymodbus.framer.rtu_framer.ModbusRtuFramer object at 0x7fd3872ad700> baud[9600])]
```

**Comment by janiversen** [comment]
In order to use the examples,you need to "git clone" the repository.

**Comment by janiversen** [comment]
please read the README, if you want to run pymodbus.console you need to install repl.

If you want to use serial you need to install serial, if you have not done so earlier.

Your code looks ok, but it cannot connect, so some of your parameters are wrong, use debug to get a bit of more information.

**Comment by jason-cso** [comment]
I did read the README.  It took me a long time to figure out that when you say `pip install -U pymodbus[<option>,…]`, the square brackets are literals.  But after figuring that out and installing `pymodbus[repl]`, running `help` doesn't list `client.connect` as an option.

I got a little further when I figured out on my own that I wasn't in the `dialout` group.  Now connect() seems to work in my code.  But how would I "use debug"?  I don't see anything in the readme about that.

**Reply by janiversen** [reply]
Being in the "dialup" group or not is not part of what we document.

Your problem with examples/helper is because you copied the example, but did not copy helper.

it's documented in our issue template (and in the general manual):
'''

# please use the following to format logs when posting them here
import pymodbus

pymodbus.pymodbus_apply_logging_config()
'''

**Comment by jason-cso** [comment]
I figured it out because pyserial makes it clear that the problem was "Permission denied".  The error message `Failed to connect[ModbusSerialClient(<pymodbus.framer.rtu_framer.ModbusRtuFramer object at 0x7fd3872ad700> baud[9600])]` is much less helpful.

I ended up using strace to watch the write() and read() calls, and it looks like my RS485 device is just echoing back my commands rather than responding appropriately.  That's on me to figure out.  But I thought you might find it useful to know my experience as a new user of your library.

**Reply by janiversen** [reply]
That is your usb not the device (typically), and there is a parameter to skip that.
